utils/helpers.py

```
@@ -0,0 +1,190 @@
+        
+    Returns:
+        Function result or raises last exception
+    """
+    for attempt in range(max_retries):
+        try:
+            return func()
+        except Exception as e:
+            if attempt == max_retries - 1:
+                raise e
+            delay = base_delay * (2 ** attempt)
+            time.sleep(delay)
```

Ruff (0.11.9)

Missing explicit `return` at the end of function able to return non-`None` value

Add explicit `return` statement

(RET503)